### ISSUE TITLE
FEAT: PyTorch Dataset 및 DataLoader 구현

### ISSUE THREAD

**Issue by coouir** [LighTrip/LighTrip-AI#82]
# FEAT: PyTorch Dataset 및 DataLoader 구현

## 목적

ROI RGB + text mask + soft label 데이터를 학습 가능한 형태로 로드하는 PyTorch Dataset과 DataLoader를 구현합니다.

## 작업 내용

- [ ] ROI 이미지 로드
- [ ] text mask 로드
- [ ] ROI RGB와 text mask를 결합해 4채널 tensor 생성
- [ ] pseudo_scores 로드
- [ ] target_distribution 로드
- [ ] train/val/test DataLoader 생성
- [ ] batch shape 검증
- [ ] 최소 augmentation 적용

## Dataset output 예시

    {
        "x": Tensor[4, 128, 256],
        "pseudo_scores": Tensor[96],
        "target_distribution": Tensor[96],
        "image_id": str
    }

## 입력 tensor 구성

    channel 1~3: ROI RGB
    channel 4: text mask

## 권장 augmentation

- horizontal flip
- weak brightness/contrast jitter

## 주의 사항

강한 색상 augmentation은 contrast label과 불일치할 수 있으므로 MVP 단계에서는 사용하지 않습니다.

## 산출물

- `src/data/dataset.py`
- `src/data/dataloader.py`

## 완료 기준

- [ ] DataLoader batch가 정상 생성된다.
- [ ] `x.shape == [B, 4, 128, 256]`이다.
- [ ] `target_distribution.shape == [B, 96]`이다.
- [ ] target distribution의 합이 1이다.
- [ ] train/val/test split이 올바르게 로드된다.

---
